app/questionnaire/questionnaire_manager.py

```diff
@@ -56,7 +56,7 @@ def validate_all_answers(self):
 
         return True, None
 
-    def update_questionnaire_store(self, location):
+    def update_questionnaire_store(self, location, save=True):
```

Comment:
Maybe, but then we'd be calling save twice - once as part of this update method and once in the new save method.